core/templates/pages/exploration-player-page/services/exploration-player-state.service.ts

```diff
@@ -152,6 +152,9 @@ angular.module('oppia').factory('ExplorationPlayerStateService', [
         var explorationData = combinedData[0];
         var featuresData = combinedData[1];
         if (doesMathExpressionInputInteractionExist(explorationData.states)) {
+          // @ts-ignore
+          // Type system does not support constructor properties (Guppy.init)
+          // thus we need ts-ignore it.
```

Comment:
Hi @nishantwrp, just a general question, can you explain why init can't be added like other properties in `guppy-defs.d.ts` for the guppy class?

Reply:
Hi Can you refer this comment. https://github.com/oppia/oppia/pull/9278#discussion_r422500557

Reply:
Hi @nishantwrp, I did take a look at that comment. That didn't answer my question. You need not repeat yourself, my specific question was, why can't it be added in the definitions file like other properties.

Reply:
Because `init` is not a property of a object belonging to class `Guppy`. It is a property of class constructor. As in
```typescript
var a = new Guppy();
// We don't want
a.init; // Declaring type for this would be possible
// but we want
Guppy.init 
```

 I don't think there is anything like that in typescript.

Reply:
Okay I looked into this a bit more and you should be able to add definitions on class methods [TypeScript handbook namespace](https://www.typescriptlang.org/docs/handbook/declaration-files/deep-dive.html#adding-using-a-namespace)

Reply:
Thanks a lot! Done!